File: Frontend/instruments-manage/src/adapters/brand.adapter.ts
import { BrandDomain } from "@/interfaces/brand-domain.interface";
import { BrandDTO } from "@/models";

export const BrandAdapter = (brands: BrandDTO): BrandDomain => {
  const { id, name, country, website } = brands;

  return {
    id,
    name,
    country,
    website,
  };
};

File: Backend/InstrumentsService/src/main/java/io/github/juniorcorzo/InstrumentsService/brands/adapter/BrandAdapter.java
package io.github.juniorcorzo.InstrumentsService.brands.adapter;

import io.github.juniorcorzo.InstrumentsService.brands.dtos.BrandsDTO;
import io.github.juniorcorzo.InstrumentsService.brands.entity.BrandsEntity;

public class BrandAdapter {
    public static BrandsEntity toEntity(BrandsDTO dto) {
        return BrandsEntity.builder()
                ._id(dto.id())
                .name(dto.name())
                .country(dto.country())
                .website(dto.website())
                .build();
    }
    
    public static BrandsDTO toDTO(BrandsEntity entity) {
        return BrandsDTO.builder()
                .id(entity.get_id())
                .name(entity.getName())
                .country(entity.getCountry())
                .website(entity.getWebsite())
                .build();
    }
}
